### PR TITLE
Add documentation for HTTP multithreaded S3 download demo

### PR DESCRIPTION
**Pull request by archigup**
This demo downloads a file from S3 using a separate thread for servicing
HTTP requests.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: docs/doxygen/demos/http_demo.dox [hunk 1]
```diff
@@ -61,4 +61,33 @@ message at least once.
...
+server's certificate is verified using its respective root CA certificate. If a
+connection failure occurs, then the connection establishment is retried for
+@ref MAX_RETRY_ATTEMPTS.
+</p>
+
+<p>
+Once a connection is established, an HTTP thread is
+started which makes HTTP requests over the TLS connection. The main thread
+loops between enqueing range requests for the S3 file on the request queue and
+dequeing repsonses from the respons queue as they are ready.  The responses
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:82]:
```suggestion
dequeing repsonses from the response queue as they are ready.  The responses
```

File: docs/doxygen/demos/http_demo.dox [hunk 2]
```diff
@@ -61,4 +61,33 @@ message at least once.
...
+from each of the requests are logged. The loop continues until the entire file
+has been requested and reveived. The HTTP thread services requests from the
+request queue and writes the responses to the respose queue.
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:85]:
```suggestion
request queue and writes the responses to the response queue.
```

**Comment by sarenameas** [comment]
image

I think these parts should these parts be in a loop block as well? Looping for until the entire range is download?

**Review by sarenameas — DISMISSED** [review]

File: docs/doxygen/images/http_demo_s3_download_multithreaded.pu [hunk 3]
```diff
@@ -0,0 +1,80 @@
+@startuml
+skinparam classFontSize 8
```
Comment (sarenameas) [inline docs/doxygen/images/http_demo_s3_download_multithreaded.pu:2]: One last nitpick can you add `skinparam dpi 300` to this image so it's not blurry on large monitors

**Review by sarenameas — DISMISSED** [review]

**Review by n9wxu — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**